File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/sound/StubPlayerEventsListener.java
/**
 *
 * @file
 *
 * @brief Stub singleton implementation of PlayerEventsListener
 *
 * @author vitamin.caig@gmail.com
 *
 */

package app.zxtune.sound;

public class StubPlayerEventsListener implements PlayerEventsListener {

  // permit inheritance
  protected StubPlayerEventsListener() {}
  
  @Override
  public void onStart() {}

  @Override
  public void onFinish() {}

  @Override
  public void onStop() {}
  
  @Override
  public void onError(Exception e) {}

  public static PlayerEventsListener instance() {
    return Holder.INSTANCE;
  }

  //onDemand holder idiom
  private static class Holder {
    public static final PlayerEventsListener INSTANCE = new StubPlayerEventsListener();
  }
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/sound/StubPlayerEventsListener.kt
/**
 * @file
 * @brief Stub implementation of PlayerEventsListener
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.sound

open class StubPlayerEventsListener  // permit inheritance
protected constructor() : PlayerEventsListener {
  override fun onStart() {}
  override fun onSeeking() {}
  override fun onFinish() {}
  override fun onStop() {}
  override fun onError(e: Exception) {}
}
